pkg/plugin/plugin.go

```diff
@@ -82,13 +82,18 @@ func (instance *StartServer[T]) Call(plugin *plugin.Plugin[PluginState], request
 		Handler: router,
 	}
 
-	go func() (map[string]any, error) {
+	var serverError error
+
+	go func() {
 		// service connections
 		if err := plugin.State.Server.ListenAndServe(); err != nil {
-			return nil, fmt.Errorf("gin framework error: %s", err)
+			serverError = fmt.Errorf("gin framework error: %s", err)
 		}
-		return nil, nil
+		serverError = nil
 	}()
```

Comment:
Can you add a comment on how you are doing this? I think for debug reason but this do not make sense if I do not know why